wgpu-hal/src/dx12/device.rs

```diff
@@ -1952,20 +1953,78 @@ impl crate::Device for super::Device {
         };
         let flags = Direct3D12::D3D12_PIPELINE_STATE_FLAG_NONE;
 
-        let raw: Direct3D12::ID3D12PipelineState = match &desc.vertex_processor {
+        let mut view_instancing: ArrayVec<Direct3D12::D3D12_VIEW_INSTANCE_LOCATION, 32> =
+            ArrayVec::new();
+        if let Some(mask) = desc.multiview_mask {
+            let mask = mask.get();
+            // The mask gets applied later
+            for i in 0..32 - mask.leading_zeros() {
+                view_instancing.push(Direct3D12::D3D12_VIEW_INSTANCE_LOCATION {
+                    ViewportArrayIndex: 0,
+                    RenderTargetArrayIndex: i,
+                });
+            }
+        }
+
+        let mut stream_desc = super::RenderPipelineStateStreamDesc {
+            // Shared by vertex and mesh pipelines
+            root_signature: root_signature
+                .as_ref()
+                .map(|a| a.as_raw().cast())
+                .unwrap_or(ptr::null_mut()),
+            pixel_shader,
+            blend_state,
+            sample_mask: desc.multisample.mask as u32,
+            rasterizer_state,
+            depth_stencil_state,
+            primitive_topology_type: topology_class,
+            rtv_formats: Direct3D12::D3D12_RT_FORMAT_ARRAY {
+                RTFormats: rtv_formats,
+                NumRenderTargets: desc.color_targets.len() as u32,
+            },
+            dsv_format,
+            sample_desc,
+            node_mask: 0,
+            cached_pso,
+            flags,
+            view_instancing: if !view_instancing.is_empty() {
+                Some(Direct3D12::D3D12_VIEW_INSTANCING_DESC {
+                    // TODO: rethink this
+                    ViewInstanceCount: view_instancing.len() as u32,
+                    pViewInstanceLocations: view_instancing.as_ptr(),
+                    // This lets us hide certain values later
+                    Flags: Direct3D12::D3D12_VIEW_INSTANCING_FLAG_ENABLE_VIEW_INSTANCE_MASKING,
+                })
+            } else {
+                None
+            },
+
+            // Other crap
```

Comment:
language! also not helpful. It's not even like this is the rest of the values that's defaulted, there's `stream_outut` in here